src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -567,6 +568,11 @@ public DatabaseRecord ReadDatabase(TransactionOperationContext context, string n
             var doc = Read(context, "db/" + name.ToLowerInvariant(), out etag);
             if (doc == null)
                 return null;
+            // If we are in passive state, we have prevent from tasks to be performed by this node.
+            if (_parent.CurrentState == RachisConsensus.State.Passive)
```

Comment:
That is extremely invasive operation here. You are basically saying that while the node is passive, no databases can be accessed.
If I just remove a node from the cluster, I lost access to all its data.